create.ps1

```diff
@@ -445,149 +396,111 @@ try {
-                    Uri         = "$($config.BaseUrl)/api/v2/hrm/employees/$($aRef)"
-                    Method      = 'PUT'
-                    Headers     = $headers
-                    Body        = ($account | ConvertTo-Json)
-                    ContentType = 'application/json; charset=utf-8'
-                }
-                $null = Invoke-RestMethod @splatEmployeeUpdateParams -Verbose:$false
+                # Only required when you do need values from the target system, for example Account Reference.
+                # Otherwise $outputContext.Data is automatically filled the with account Data
+                $outputContext.Data = $createdAccount
+                $accountReferenceId = $createdAccount.id
+                $auditLogMessage = "Create account was successful. AccountReference is: [$($outputContext.AccountReference)"
                 break
             }
 
-            'Correlate' {
-                if ($null -eq (Compare-Object $account.positionsPerOrgUnits   $employee.positionsPerOrgUnits)) {
-                    Write-Verbose 'Correlating Inception Employee account'
-                    Write-Verbose  'No position update required'
-                }
-                else {
-                    Write-Verbose 'Updating Position and correlating Inception Employee account'
-                    $splatEmployeeUpdateParams = @{
-                        Uri         = "$($config.BaseUrl)/api/v2/hrm/employees/$($aRef)"
-                        Method      = 'PUT'
-                        Headers     = $headers
-                        Body        = (($account | Select-Object positionsPerOrgUnits ) | ConvertTo-Json)
-                        ContentType = 'application/json; charset=utf-8'
-                    }
-                    $null = Invoke-RestMethod @splatEmployeeUpdateParams -Verbose:$false
-                }
+            'CorrelateAccount' {
+                Write-Verbose 'Correlating Inception employee account'
+                $correlatedAccount = $correlatedAccount | Select-Object -Property * -ExcludeProperty $excludedPropertiesArray
+                $outputContext.Data = $correlatedAccount
+                $accountReferenceId = $correlatedAccount.id
+                $outputContext.AccountCorrelated = $true
+                $auditLogMessage = "Correlated account: [$($correlatedAccount.id)] on field: [$($correlationField)] with value: [$($correlationValue)]"
                 break
             }
         }
-    }
-    $success = $true
-    $auditLogs.Add([PSCustomObject]@{
-            Message = "$action Employee account was successful. AccountReference is: [$aRef]"
-            IsError = $false
-        })
 
+        $accountReferenceObject = @{
+            AccountReference = $accountReferenceId
+            Positions        = $desiredPositionList | Select-Object * -ExcludeProperty SideIndicator
+        }
+
+        $outputContext.AccountReference = $accountReferenceObject
+        $outputContext.success = $true
```

Comment:
DryRun will never make success = $true.

I recommend adding a finaly to your scripts and starting the script with success = $true. For example:

```Powershell
finally {
    # Check if auditLogs contains errors, if errors are found, set success to false
    if ($outputContext.AuditLogs.IsError -contains $true) {
        $outputContext.Success = $false
    }
}
```